src/UI/controllers/OverlayUIController.js

```diff
@@ -77,19 +77,20 @@ export class OverlayUIController extends UIController {
 
         const position = getPosition()
 
-        let delay = 0
         if (!click && !this.elementIsInViewport(position)) {
             input.scrollIntoView(true)
-            delay = 500
+            this._mutObs?.disconnect()
+            setTimeout(() => {
+                this.attach(args)
+            }, 50)
+            return
```

Comment:
Instead of having the side-effect happen after a timeout, now we are calling this same function recursively after the timeout. This ensures that `this.#state` is consistent across calls. The `500` delay was likely arbitrary and `50` works just as well, so shorter is better.

Reply:
@GioSensation NAB: Do we want to provide a comment for context on the `50` value to help future  us? (i.e. it's just some value to throttle repeated calls)

Reply:
I've added a comment above to explain why we're recursing, plus that the 50 is arbitrary.